src/foam/nanos/ruler/Rule.js

```diff
@@ -275,9 +275,8 @@
           try {
             return getPredicate().f(obj);
           } catch ( Throwable th ) { }
-
-          ((Logger) x.get("logger")).error(
-            "Failed to evaluate predicate of rule: " + getId(), t);
+          // ((Logger) x.get("logger")).debug(this.getClass().getSimpleName(), "id", getId(), "\\nrule", this, "\\nobj", obj, "\\nold", oldObj, "\\n", t);
```

Comment:
do we want to leave the commented code here?

Reply:
yes. debugging rules is tricky, and keep enabling these to see what is going on.